src/spaces/utilities.jl

```diff
@@ -233,6 +235,158 @@ walk!(agent, ::typeof(rand), model::ABM{<:AbstractGridSpace{D}}; kwargs...) wher
 walk!(agent, ::typeof(rand), model::ABM{<:ContinuousSpace{D}}) where {D} =
     walk!(agent, Tuple(2.0 * rand(model.rng) - 1.0 for _ in 1:D), model)
 
+"""
+    randomwalk!(agent, model::ABM{<:AbstractGridSpace}, r; kwargs...)
+Move `agent` for a distance `r` in a random direction respecting boundary conditions
+and space metric.
+For Chebyshev and Manhattan metric, the step size `r` is rounded to `floor(Int,r)`;
+for Euclidean metric in a GridSpace, random walks are not defined due to a strong
+dependency on the given value of `r`.
+
+## Keywords
+- `ifempty` will check that the target position is unoccupied and only move if that's true.
+  By default this is true, set it to false if different agents can occupy the same position.
+  In a GridSpaceSingle, agents cannot overlap and this keyword has no effect.
+"""
+function randomwalk!(
+    agent::AbstractAgent,
+    model::ABM{<:AbstractGridSpace},
+    r::Real;
+    kwargs...
+)
+    if model.space.metric == :euclidean
+        throw(ArgumentError(
+            "Random walks on a GridSpace with Euclidean metric are not defined. " *
+            "You might want to use a ContinuousSpace or a different metric."
```

Comment:
```suggestion
            "Random walks on a `GridSpace` with Euclidean metric are not defined. " *
            "You might want to use a `ContinuousSpace` or a different metric."
```
Error messages are like comments or like docstrings. They support markdown syntax. When referring to code objects, use backticks